File: backend-user/src/main/java/com/matchpet/backend_user/dto/UpdateAdoptanteRequest.java
package com.matchpet.backend_user.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UpdateAdoptanteRequest {

    // --- Datos de UserModel ---
    @NotBlank(message = "El nombre completo es requerido")
    private String nombreCompleto;

    @NotBlank(message = "El teléfono es requerido")
    private String telefono;

    // --- Datos de PerfilAdoptante ---
    private Date fechaNacimiento; // Formato: "YYYY-MM-DD"
    private String direccion;
    private String ciudad;

    // TODO: Aquí se podrían añadir las "preferencias de mascota" de la HU-05
}

File: frontend-movile/app/src/main/java/com/example/matchpet/data/model/adoptante/UpdateAdoptanteRequest.kt
package com.example.matchpet.data.model.adoptante

data class UpdateAdoptanteRequest(
    val nombre: String,
    val apellidoPaterno: String,
    val apellidoMaterno: String,
    val telefono: String,
    val fechaNacimiento: String, // Formato YYYY-MM-DD
    val direccion: String,
    val ciudad: String,
    val pais: String
)